python/pyspark/shell.py

```diff
@@ -76,4 +76,6 @@
 # which allows us to execute the user's PYTHONSTARTUP file:
 _pythonstartup = os.environ.get('OLD_PYTHONSTARTUP')
 if _pythonstartup and os.path.isfile(_pythonstartup):
-    execfile(_pythonstartup)
+    with open(_pythonstartup) as f:
```

Comment:
One minor consequence of this change is that the variable `f` is now defined in in the REPL, e.g. `f` will now point to a closed file. We could try to add some fancy logic to prevent this, but it would be equally brittle in the opposite direction. Therefore I'm fine with this change; it's extremely unlikely that user code in the REPL will take conditional actions based on whether `f` is defined.